src/App.tsx

```diff
@@ -14,11 +39,99 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
+
+type State = {
+  listOfProducts: string[];
+  selectedProducts: string[];
+};
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    listOfProducts: goodsFromServer,
+    selectedProducts: ['Jam'],
+  };
+
+  addProduct = (product: string) => {
+    this.setState((prevState) => ({
+      selectedProducts: [...prevState.selectedProducts, product],
+    }));
+  };
+
+  removeProduct = (product: string) => {
+    this.setState((prevState) => {
+      const { selectedProducts } = prevState;
+      const index = selectedProducts.indexOf(product);
+      const newProducts = selectedProducts
+        .slice(0, index)
+        .concat(selectedProducts.slice(index + 1));
+
+      return ({
+        selectedProducts: [...newProducts],
+      });
+    });
+  };
+
+  clearProducts = () => {
+    this.setState({ selectedProducts: [] });
+  };
+
+  render() {
+    const { listOfProducts, selectedProducts } = this.state;
+
+    return (
+      <div className="App">
+        <div className="list container box">
+          <ul>
+            {listOfProducts.map((product) => (
+              <li
+                key={product}
+                className={`list__item ${selectedProducts.includes(product) ? 'list__item--selected' : ''}`}
```

Comment:
giving empty string a ternary result is not the best solution, for sure
you could try this one, for example: https://www.npmjs.com/package/classnames
it is frequently used in real projects